// gate-api/dist/api/apiClient.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ApiClient = void 0;
const models_1 = require("../model/models");
const json_bigint_1 = __importDefault(require("json-bigint"));
const axios_1 = __importDefault(require("axios"));
class ApiClient {
    constructor(basePath, axiosInstance = axios_1.default) {
        this.axiosInstance = axiosInstance;
        this._basePath = 'https://api.gateio.ws/api/v4';
        this._defaultHeaders = {};
        this.authentications = {
            apiv4: new models_1.GateApiV4Auth(),
        };
        this._basePath = basePath || this._basePath;
        this.axiosInstance.defaults.transformResponse = [
            (data) => {
                try {
                    return json_bigint_1.default.parse(data);
                }
                catch (error) {
                    console.error('Failed to parse JSON:', error);
                    return data;
                }
            },
        ];
    }
    set basePath(basePath) {
        this._basePath = basePath;
    }
    get basePath() {
        return this._basePath;
    }
    set defaultHeaders(defaultHeaders) {
        this._defaultHeaders = defaultHeaders;
    }
    get defaultHeaders() {
        return this._defaultHeaders;
    }
    setApiKeySecret(key, secret) {
        let auth = this.authentications['apiv4'];
        auth.key = key;
        auth.secret = secret;
    }
    applyToRequest(config, authSettings) {
        for (let auth of authSettings) {
            let authenticator = this.authentications[auth];
            if (authenticator) {
                config = authenticator.applyToRequest(config);
            }
        }
        return config;
    }
    async request(config, responseType, authSettings) {
        return Promise.resolve(config)
            .then((c) => this.applyToRequest(c, authSettings))
            .then((c) => {
            return this.axiosInstance.request(c).then((rsp) => {
                let body = rsp.data;
                if (responseType.length > 0) {
                    body = models_1.ObjectSerializer.deserialize(rsp.data, responseType);
                }
                return { response: rsp, body: body };
            });
        });
    }
}
exports.ApiClient = ApiClient;
