// gate-api/dist/api/withdrawalApi.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WithdrawalApi = void 0;
const models_1 = require("../model/models");
const apiClient_1 = require("./apiClient");
class WithdrawalApi {
    constructor(client) {
        if (client) {
            this.client = client;
        }
        else {
            this.client = new apiClient_1.ApiClient();
        }
    }
    async withdraw(ledgerRecord) {
        const localVarPath = this.client.basePath + '/withdrawals';
        let localVarQueryParameters = {};
        let localVarHeaderParams = Object.assign({}, this.client.defaultHeaders);
        const produces = ['application/json'];
        if (produces.indexOf('application/json') >= 0) {
            localVarHeaderParams.Accept = 'application/json';
        }
        else {
            localVarHeaderParams.Accept = produces.join(',');
        }
        if (ledgerRecord === null || ledgerRecord === undefined) {
            throw new Error('Required parameter ledgerRecord was null or undefined when calling withdraw.');
        }
        const config = {
            method: 'POST',
            params: localVarQueryParameters,
            headers: localVarHeaderParams,
            url: localVarPath,
            data: models_1.ObjectSerializer.serialize(ledgerRecord, 'LedgerRecord'),
        };
        const authSettings = ['apiv4'];
        return this.client.request(config, 'LedgerRecord', authSettings);
    }
    async withdrawPushOrder(uidPushWithdrawal) {
        const localVarPath = this.client.basePath + '/withdrawals/push';
        let localVarQueryParameters = {};
        let localVarHeaderParams = Object.assign({}, this.client.defaultHeaders);
        const produces = ['application/json'];
        if (produces.indexOf('application/json') >= 0) {
            localVarHeaderParams.Accept = 'application/json';
        }
        else {
            localVarHeaderParams.Accept = produces.join(',');
        }
        if (uidPushWithdrawal === null || uidPushWithdrawal === undefined) {
            throw new Error('Required parameter uidPushWithdrawal was null or undefined when calling withdrawPushOrder.');
        }
        const config = {
            method: 'POST',
            params: localVarQueryParameters,
            headers: localVarHeaderParams,
            url: localVarPath,
            data: models_1.ObjectSerializer.serialize(uidPushWithdrawal, 'UidPushWithdrawal'),
        };
        const authSettings = ['apiv4'];
        return this.client.request(config, 'UidPushWithdrawalResp', authSettings);
    }
    async cancelWithdrawal(withdrawalId) {
        const localVarPath = this.client.basePath +
            '/withdrawals/{withdrawal_id}'.replace('{' + 'withdrawal_id' + '}', encodeURIComponent(String(withdrawalId)));
        let localVarQueryParameters = {};
        let localVarHeaderParams = Object.assign({}, this.client.defaultHeaders);
        const produces = ['application/json'];
        if (produces.indexOf('application/json') >= 0) {
            localVarHeaderParams.Accept = 'application/json';
        }
        else {
            localVarHeaderParams.Accept = produces.join(',');
        }
        if (withdrawalId === null || withdrawalId === undefined) {
            throw new Error('Required parameter withdrawalId was null or undefined when calling cancelWithdrawal.');
        }
        const config = {
            method: 'DELETE',
            params: localVarQueryParameters,
            headers: localVarHeaderParams,
            url: localVarPath,
        };
        const authSettings = ['apiv4'];
        return this.client.request(config, 'LedgerRecord', authSettings);
    }
}
exports.WithdrawalApi = WithdrawalApi;
