// gate-api/dist/model/accountBalance.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountBalance = void 0;
class AccountBalance {
    static getAttributeTypeMap() {
        return AccountBalance.attributeTypeMap;
    }
}
exports.AccountBalance = AccountBalance;
AccountBalance.discriminator = undefined;
AccountBalance.attributeTypeMap = [
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'currency',
        baseName: 'currency',
        type: 'AccountBalance.Currency',
    },
    {
        name: 'unrealisedPnl',
        baseName: 'unrealised_pnl',
        type: 'string',
    },
    {
        name: 'borrowed',
        baseName: 'borrowed',
        type: 'string',
    },
];
(function (AccountBalance) {
    let Currency;
    (function (Currency) {
        Currency[Currency["BTC"] = 'BTC'] = "BTC";
        Currency[Currency["CNY"] = 'CNY'] = "CNY";
        Currency[Currency["USD"] = 'USD'] = "USD";
        Currency[Currency["USDT"] = 'USDT'] = "USDT";
    })(Currency = AccountBalance.Currency || (AccountBalance.Currency = {}));
})(AccountBalance || (exports.AccountBalance = AccountBalance = {}));
