// gate-api/dist/model/batchOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchOrder = void 0;
class BatchOrder {
    static getAttributeTypeMap() {
        return BatchOrder.attributeTypeMap;
    }
}
exports.BatchOrder = BatchOrder;
BatchOrder.discriminator = undefined;
BatchOrder.attributeTypeMap = [
    {
        name: 'orderId',
        baseName: 'order_id',
        type: 'string',
    },
    {
        name: 'amendText',
        baseName: 'amend_text',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'succeeded',
        baseName: 'succeeded',
        type: 'boolean',
    },
    {
        name: 'label',
        baseName: 'label',
        type: 'string',
    },
    {
        name: 'message',
        baseName: 'message',
        type: 'string',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'string',
    },
    {
        name: 'updateTime',
        baseName: 'update_time',
        type: 'string',
    },
    {
        name: 'createTimeMs',
        baseName: 'create_time_ms',
        type: 'number',
    },
    {
        name: 'updateTimeMs',
        baseName: 'update_time_ms',
        type: 'number',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'BatchOrder.Status',
    },
    {
        name: 'currencyPair',
        baseName: 'currency_pair',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'BatchOrder.Type',
    },
    {
        name: 'account',
        baseName: 'account',
        type: 'BatchOrder.Account',
    },
    {
        name: 'side',
        baseName: 'side',
        type: 'BatchOrder.Side',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'timeInForce',
        baseName: 'time_in_force',
        type: 'BatchOrder.TimeInForce',
    },
    {
        name: 'iceberg',
        baseName: 'iceberg',
        type: 'string',
    },
    {
        name: 'autoBorrow',
        baseName: 'auto_borrow',
        type: 'boolean',
    },
    {
        name: 'autoRepay',
        baseName: 'auto_repay',
        type: 'boolean',
    },
    {
        name: 'left',
        baseName: 'left',
        type: 'string',
    },
    {
        name: 'filledAmount',
        baseName: 'filled_amount',
        type: 'string',
    },
    {
        name: 'fillPrice',
        baseName: 'fill_price',
        type: 'string',
    },
    {
        name: 'filledTotal',
        baseName: 'filled_total',
        type: 'string',
    },
    {
        name: 'avgDealPrice',
        baseName: 'avg_deal_price',
        type: 'string',
    },
    {
        name: 'fee',
        baseName: 'fee',
        type: 'string',
    },
    {
        name: 'feeCurrency',
        baseName: 'fee_currency',
        type: 'string',
    },
    {
        name: 'pointFee',
        baseName: 'point_fee',
        type: 'string',
    },
    {
        name: 'gtFee',
        baseName: 'gt_fee',
        type: 'string',
    },
    {
        name: 'gtDiscount',
        baseName: 'gt_discount',
        type: 'boolean',
    },
    {
        name: 'rebatedFee',
        baseName: 'rebated_fee',
        type: 'string',
    },
    {
        name: 'rebatedFeeCurrency',
        baseName: 'rebated_fee_currency',
        type: 'string',
    },
    {
        name: 'stpId',
        baseName: 'stp_id',
        type: 'number',
    },
    {
        name: 'stpAct',
        baseName: 'stp_act',
        type: 'BatchOrder.StpAct',
    },
    {
        name: 'finishAs',
        baseName: 'finish_as',
        type: 'BatchOrder.FinishAs',
    },
];
(function (BatchOrder) {
    let Status;
    (function (Status) {
        Status[Status["Open"] = 'open'] = "Open";
        Status[Status["Closed"] = 'closed'] = "Closed";
        Status[Status["Cancelled"] = 'cancelled'] = "Cancelled";
    })(Status = BatchOrder.Status || (BatchOrder.Status = {}));
    let Type;
    (function (Type) {
        Type[Type["Limit"] = 'limit'] = "Limit";
        Type[Type["Market"] = 'market'] = "Market";
    })(Type = BatchOrder.Type || (BatchOrder.Type = {}));
    let Account;
    (function (Account) {
        Account[Account["Spot"] = 'spot'] = "Spot";
        Account[Account["Margin"] = 'margin'] = "Margin";
        Account[Account["CrossMargin"] = 'cross_margin'] = "CrossMargin";
        Account[Account["Unified"] = 'unified'] = "Unified";
    })(Account = BatchOrder.Account || (BatchOrder.Account = {}));
    let Side;
    (function (Side) {
        Side[Side["Buy"] = 'buy'] = "Buy";
        Side[Side["Sell"] = 'sell'] = "Sell";
    })(Side = BatchOrder.Side || (BatchOrder.Side = {}));
    let TimeInForce;
    (function (TimeInForce) {
        TimeInForce[TimeInForce["Gtc"] = 'gtc'] = "Gtc";
        TimeInForce[TimeInForce["Ioc"] = 'ioc'] = "Ioc";
        TimeInForce[TimeInForce["Poc"] = 'poc'] = "Poc";
        TimeInForce[TimeInForce["Fok"] = 'fok'] = "Fok";
    })(TimeInForce = BatchOrder.TimeInForce || (BatchOrder.TimeInForce = {}));
    let StpAct;
    (function (StpAct) {
        StpAct[StpAct["Cn"] = 'cn'] = "Cn";
        StpAct[StpAct["Co"] = 'co'] = "Co";
        StpAct[StpAct["Cb"] = 'cb'] = "Cb";
        StpAct[StpAct["Minus"] = '-'] = "Minus";
    })(StpAct = BatchOrder.StpAct || (BatchOrder.StpAct = {}));
    let FinishAs;
    (function (FinishAs) {
        FinishAs[FinishAs["Open"] = 'open'] = "Open";
        FinishAs[FinishAs["Filled"] = 'filled'] = "Filled";
        FinishAs[FinishAs["Cancelled"] = 'cancelled'] = "Cancelled";
        FinishAs[FinishAs["Ioc"] = 'ioc'] = "Ioc";
        FinishAs[FinishAs["Stp"] = 'stp'] = "Stp";
    })(FinishAs = BatchOrder.FinishAs || (BatchOrder.FinishAs = {}));
})(BatchOrder || (exports.BatchOrder = BatchOrder = {}));
