// gate-api/dist/model/batchFuturesOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BatchFuturesOrder = void 0;
class BatchFuturesOrder {
    static getAttributeTypeMap() {
        return BatchFuturesOrder.attributeTypeMap;
    }
}
exports.BatchFuturesOrder = BatchFuturesOrder;
BatchFuturesOrder.discriminator = undefined;
BatchFuturesOrder.attributeTypeMap = [
    {
        name: 'succeeded',
        baseName: 'succeeded',
        type: 'boolean',
    },
    {
        name: 'label',
        baseName: 'label',
        type: 'string',
    },
    {
        name: 'detail',
        baseName: 'detail',
        type: 'string',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'number',
    },
    {
        name: 'user',
        baseName: 'user',
        type: 'number',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'finishTime',
        baseName: 'finish_time',
        type: 'number',
    },
    {
        name: 'finishAs',
        baseName: 'finish_as',
        type: 'BatchFuturesOrder.FinishAs',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'BatchFuturesOrder.Status',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'iceberg',
        baseName: 'iceberg',
        type: 'number',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'close',
        baseName: 'close',
        type: 'boolean',
    },
    {
        name: 'isClose',
        baseName: 'is_close',
        type: 'boolean',
    },
    {
        name: 'reduceOnly',
        baseName: 'reduce_only',
        type: 'boolean',
    },
    {
        name: 'isReduceOnly',
        baseName: 'is_reduce_only',
        type: 'boolean',
    },
    {
        name: 'isLiq',
        baseName: 'is_liq',
        type: 'boolean',
    },
    {
        name: 'tif',
        baseName: 'tif',
        type: 'BatchFuturesOrder.Tif',
    },
    {
        name: 'left',
        baseName: 'left',
        type: 'number',
    },
    {
        name: 'fillPrice',
        baseName: 'fill_price',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'tkfr',
        baseName: 'tkfr',
        type: 'string',
    },
    {
        name: 'mkfr',
        baseName: 'mkfr',
        type: 'string',
    },
    {
        name: 'refu',
        baseName: 'refu',
        type: 'number',
    },
    {
        name: 'autoSize',
        baseName: 'auto_size',
        type: 'BatchFuturesOrder.AutoSize',
    },
    {
        name: 'stpAct',
        baseName: 'stp_act',
        type: 'BatchFuturesOrder.StpAct',
    },
    {
        name: 'stpId',
        baseName: 'stp_id',
        type: 'number',
    },
];
(function (BatchFuturesOrder) {
    let FinishAs;
    (function (FinishAs) {
        FinishAs[FinishAs["Filled"] = 'filled'] = "Filled";
        FinishAs[FinishAs["Cancelled"] = 'cancelled'] = "Cancelled";
        FinishAs[FinishAs["Liquidated"] = 'liquidated'] = "Liquidated";
        FinishAs[FinishAs["Ioc"] = 'ioc'] = "Ioc";
        FinishAs[FinishAs["AutoDeleveraged"] = 'auto_deleveraged'] = "AutoDeleveraged";
        FinishAs[FinishAs["ReduceOnly"] = 'reduce_only'] = "ReduceOnly";
        FinishAs[FinishAs["PositionClosed"] = 'position_closed'] = "PositionClosed";
        FinishAs[FinishAs["ReduceOut"] = 'reduce_out'] = "ReduceOut";
        FinishAs[FinishAs["Stp"] = 'stp'] = "Stp";
    })(FinishAs = BatchFuturesOrder.FinishAs || (BatchFuturesOrder.FinishAs = {}));
    let Status;
    (function (Status) {
        Status[Status["Open"] = 'open'] = "Open";
        Status[Status["Finished"] = 'finished'] = "Finished";
    })(Status = BatchFuturesOrder.Status || (BatchFuturesOrder.Status = {}));
    let Tif;
    (function (Tif) {
        Tif[Tif["Gtc"] = 'gtc'] = "Gtc";
        Tif[Tif["Ioc"] = 'ioc'] = "Ioc";
        Tif[Tif["Poc"] = 'poc'] = "Poc";
        Tif[Tif["Fok"] = 'fok'] = "Fok";
    })(Tif = BatchFuturesOrder.Tif || (BatchFuturesOrder.Tif = {}));
    let AutoSize;
    (function (AutoSize) {
        AutoSize[AutoSize["Long"] = 'close_long'] = "Long";
        AutoSize[AutoSize["Short"] = 'close_short'] = "Short";
    })(AutoSize = BatchFuturesOrder.AutoSize || (BatchFuturesOrder.AutoSize = {}));
    let StpAct;
    (function (StpAct) {
        StpAct[StpAct["Co"] = 'co'] = "Co";
        StpAct[StpAct["Cn"] = 'cn'] = "Cn";
        StpAct[StpAct["Cb"] = 'cb'] = "Cb";
        StpAct[StpAct["Minus"] = '-'] = "Minus";
    })(StpAct = BatchFuturesOrder.StpAct || (BatchFuturesOrder.StpAct = {}));
})(BatchFuturesOrder || (exports.BatchFuturesOrder = BatchFuturesOrder = {}));
