// gate-api/dist/model/createUniLend.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUniLend = void 0;
class CreateUniLend {
    static getAttributeTypeMap() {
        return CreateUniLend.attributeTypeMap;
    }
}
exports.CreateUniLend = CreateUniLend;
CreateUniLend.discriminator = undefined;
CreateUniLend.attributeTypeMap = [
    {
        name: 'currency',
        baseName: 'currency',
        type: 'string',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'CreateUniLend.Type',
    },
    {
        name: 'minRate',
        baseName: 'min_rate',
        type: 'string',
    },
];
(function (CreateUniLend) {
    let Type;
    (function (Type) {
        Type[Type["Lend"] = 'lend'] = "Lend";
        Type[Type["Redeem"] = 'redeem'] = "Redeem";
    })(Type = CreateUniLend.Type || (CreateUniLend.Type = {}));
})(CreateUniLend || (exports.CreateUniLend = CreateUniLend = {}));
