// gate-api/dist/model/createUniLoan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUniLoan = void 0;
class CreateUniLoan {
    static getAttributeTypeMap() {
        return CreateUniLoan.attributeTypeMap;
    }
}
exports.CreateUniLoan = CreateUniLoan;
CreateUniLoan.discriminator = undefined;
CreateUniLoan.attributeTypeMap = [
    {
        name: 'currency',
        baseName: 'currency',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'CreateUniLoan.Type',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'repaidAll',
        baseName: 'repaid_all',
        type: 'boolean',
    },
    {
        name: 'currencyPair',
        baseName: 'currency_pair',
        type: 'string',
    },
];
(function (CreateUniLoan) {
    let Type;
    (function (Type) {
        Type[Type["Borrow"] = 'borrow'] = "Borrow";
        Type[Type["Repay"] = 'repay'] = "Repay";
    })(Type = CreateUniLoan.Type || (CreateUniLoan.Type = {}));
})(CreateUniLoan || (exports.CreateUniLoan = CreateUniLoan = {}));
