// gate-api/dist/model/crossMarginLoan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CrossMarginLoan = void 0;
class CrossMarginLoan {
    static getAttributeTypeMap() {
        return CrossMarginLoan.attributeTypeMap;
    }
}
exports.CrossMarginLoan = CrossMarginLoan;
CrossMarginLoan.discriminator = undefined;
CrossMarginLoan.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'string',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'updateTime',
        baseName: 'update_time',
        type: 'number',
    },
    {
        name: 'currency',
        baseName: 'currency',
        type: 'string',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'CrossMarginLoan.Status',
    },
    {
        name: 'repaid',
        baseName: 'repaid',
        type: 'string',
    },
    {
        name: 'repaidInterest',
        baseName: 'repaid_interest',
        type: 'string',
    },
    {
        name: 'unpaidInterest',
        baseName: 'unpaid_interest',
        type: 'string',
    },
];
(function (CrossMarginLoan) {
    let Status;
    (function (Status) {
        Status[Status["NUMBER_1"] = 1] = "NUMBER_1";
        Status[Status["NUMBER_2"] = 2] = "NUMBER_2";
        Status[Status["NUMBER_3"] = 3] = "NUMBER_3";
    })(Status = CrossMarginLoan.Status || (CrossMarginLoan.Status = {}));
})(CrossMarginLoan || (exports.CrossMarginLoan = CrossMarginLoan = {}));
