// gate-api/dist/model/futuresPriceTriggeredOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FuturesPriceTriggeredOrder = void 0;
class FuturesPriceTriggeredOrder {
    static getAttributeTypeMap() {
        return FuturesPriceTriggeredOrder.attributeTypeMap;
    }
}
exports.FuturesPriceTriggeredOrder = FuturesPriceTriggeredOrder;
FuturesPriceTriggeredOrder.discriminator = undefined;
FuturesPriceTriggeredOrder.attributeTypeMap = [
    {
        name: 'initial',
        baseName: 'initial',
        type: 'FuturesInitialOrder',
    },
    {
        name: 'trigger',
        baseName: 'trigger',
        type: 'FuturesPriceTrigger',
    },
    {
        name: 'id',
        baseName: 'id',
        type: 'number',
    },
    {
        name: 'user',
        baseName: 'user',
        type: 'number',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'finishTime',
        baseName: 'finish_time',
        type: 'number',
    },
    {
        name: 'tradeId',
        baseName: 'trade_id',
        type: 'number',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'FuturesPriceTriggeredOrder.Status',
    },
    {
        name: 'finishAs',
        baseName: 'finish_as',
        type: 'FuturesPriceTriggeredOrder.FinishAs',
    },
    {
        name: 'reason',
        baseName: 'reason',
        type: 'string',
    },
    {
        name: 'orderType',
        baseName: 'order_type',
        type: 'string',
    },
    {
        name: 'meOrderId',
        baseName: 'me_order_id',
        type: 'number',
    },
];
(function (FuturesPriceTriggeredOrder) {
    let Status;
    (function (Status) {
        Status[Status["Open"] = 'open'] = "Open";
        Status[Status["Finished"] = 'finished'] = "Finished";
        Status[Status["Inactive"] = 'inactive'] = "Inactive";
        Status[Status["Invalid"] = 'invalid'] = "Invalid";
    })(Status = FuturesPriceTriggeredOrder.Status || (FuturesPriceTriggeredOrder.Status = {}));
    let FinishAs;
    (function (FinishAs) {
        FinishAs[FinishAs["Cancelled"] = 'cancelled'] = "Cancelled";
        FinishAs[FinishAs["Succeeded"] = 'succeeded'] = "Succeeded";
        FinishAs[FinishAs["Failed"] = 'failed'] = "Failed";
        FinishAs[FinishAs["Expired"] = 'expired'] = "Expired";
    })(FinishAs = FuturesPriceTriggeredOrder.FinishAs || (FuturesPriceTriggeredOrder.FinishAs = {}));
})(FuturesPriceTriggeredOrder || (exports.FuturesPriceTriggeredOrder = FuturesPriceTriggeredOrder = {}));
