// gate-api/dist/model/futuresOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FuturesOrder = void 0;
class FuturesOrder {
    static getAttributeTypeMap() {
        return FuturesOrder.attributeTypeMap;
    }
}
exports.FuturesOrder = FuturesOrder;
FuturesOrder.discriminator = undefined;
FuturesOrder.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'bigint',
    },
    {
        name: 'user',
        baseName: 'user',
        type: 'number',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'finishTime',
        baseName: 'finish_time',
        type: 'number',
    },
    {
        name: 'finishAs',
        baseName: 'finish_as',
        type: 'FuturesOrder.FinishAs',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'FuturesOrder.Status',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'iceberg',
        baseName: 'iceberg',
        type: 'number',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'close',
        baseName: 'close',
        type: 'boolean',
    },
    {
        name: 'isClose',
        baseName: 'is_close',
        type: 'boolean',
    },
    {
        name: 'reduceOnly',
        baseName: 'reduce_only',
        type: 'boolean',
    },
    {
        name: 'isReduceOnly',
        baseName: 'is_reduce_only',
        type: 'boolean',
    },
    {
        name: 'isLiq',
        baseName: 'is_liq',
        type: 'boolean',
    },
    {
        name: 'tif',
        baseName: 'tif',
        type: 'FuturesOrder.Tif',
    },
    {
        name: 'left',
        baseName: 'left',
        type: 'number',
    },
    {
        name: 'fillPrice',
        baseName: 'fill_price',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'tkfr',
        baseName: 'tkfr',
        type: 'string',
    },
    {
        name: 'mkfr',
        baseName: 'mkfr',
        type: 'string',
    },
    {
        name: 'refu',
        baseName: 'refu',
        type: 'number',
    },
    {
        name: 'autoSize',
        baseName: 'auto_size',
        type: 'FuturesOrder.AutoSize',
    },
    {
        name: 'stpId',
        baseName: 'stp_id',
        type: 'number',
    },
    {
        name: 'stpAct',
        baseName: 'stp_act',
        type: 'FuturesOrder.StpAct',
    },
    {
        name: 'amendText',
        baseName: 'amend_text',
        type: 'string',
    },
    {
        name: 'bizInfo',
        baseName: 'biz_info',
        type: 'string',
    },
];
(function (FuturesOrder) {
    let FinishAs;
    (function (FinishAs) {
        FinishAs[FinishAs["Filled"] = 'filled'] = "Filled";
        FinishAs[FinishAs["Cancelled"] = 'cancelled'] = "Cancelled";
        FinishAs[FinishAs["Liquidated"] = 'liquidated'] = "Liquidated";
        FinishAs[FinishAs["Ioc"] = 'ioc'] = "Ioc";
        FinishAs[FinishAs["AutoDeleveraged"] = 'auto_deleveraged'] = "AutoDeleveraged";
        FinishAs[FinishAs["ReduceOnly"] = 'reduce_only'] = "ReduceOnly";
        FinishAs[FinishAs["PositionClosed"] = 'position_closed'] = "PositionClosed";
        FinishAs[FinishAs["ReduceOut"] = 'reduce_out'] = "ReduceOut";
        FinishAs[FinishAs["Stp"] = 'stp'] = "Stp";
    })(FinishAs = FuturesOrder.FinishAs || (FuturesOrder.FinishAs = {}));
    let Status;
    (function (Status) {
        Status[Status["Open"] = 'open'] = "Open";
        Status[Status["Finished"] = 'finished'] = "Finished";
    })(Status = FuturesOrder.Status || (FuturesOrder.Status = {}));
    let Tif;
    (function (Tif) {
        Tif[Tif["Gtc"] = 'gtc'] = "Gtc";
        Tif[Tif["Ioc"] = 'ioc'] = "Ioc";
        Tif[Tif["Poc"] = 'poc'] = "Poc";
        Tif[Tif["Fok"] = 'fok'] = "Fok";
    })(Tif = FuturesOrder.Tif || (FuturesOrder.Tif = {}));
    let AutoSize;
    (function (AutoSize) {
        AutoSize[AutoSize["Long"] = 'close_long'] = "Long";
        AutoSize[AutoSize["Short"] = 'close_short'] = "Short";
    })(AutoSize = FuturesOrder.AutoSize || (FuturesOrder.AutoSize = {}));
    let StpAct;
    (function (StpAct) {
        StpAct[StpAct["Co"] = 'co'] = "Co";
        StpAct[StpAct["Cn"] = 'cn'] = "Cn";
        StpAct[StpAct["Cb"] = 'cb'] = "Cb";
        StpAct[StpAct["Minus"] = '-'] = "Minus";
    })(StpAct = FuturesOrder.StpAct || (FuturesOrder.StpAct = {}));
})(FuturesOrder || (exports.FuturesOrder = FuturesOrder = {}));
