// gate-api/dist/model/myFuturesTradeTimeRange.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MyFuturesTradeTimeRange = void 0;
class MyFuturesTradeTimeRange {
    static getAttributeTypeMap() {
        return MyFuturesTradeTimeRange.attributeTypeMap;
    }
}
exports.MyFuturesTradeTimeRange = MyFuturesTradeTimeRange;
MyFuturesTradeTimeRange.discriminator = undefined;
MyFuturesTradeTimeRange.attributeTypeMap = [
    {
        name: 'tradeId',
        baseName: 'trade_id',
        type: 'string',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'orderId',
        baseName: 'order_id',
        type: 'string',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'closeSize',
        baseName: 'close_size',
        type: 'number',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'MyFuturesTradeTimeRange.Role',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'fee',
        baseName: 'fee',
        type: 'string',
    },
    {
        name: 'pointFee',
        baseName: 'point_fee',
        type: 'string',
    },
];
(function (MyFuturesTradeTimeRange) {
    let Role;
    (function (Role) {
        Role[Role["Taker"] = 'taker'] = "Taker";
        Role[Role["Maker"] = 'maker'] = "Maker";
    })(Role = MyFuturesTradeTimeRange.Role || (MyFuturesTradeTimeRange.Role = {}));
})(MyFuturesTradeTimeRange || (exports.MyFuturesTradeTimeRange = MyFuturesTradeTimeRange = {}));
