// gate-api/dist/model/optionsOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionsOrder = void 0;
class OptionsOrder {
    static getAttributeTypeMap() {
        return OptionsOrder.attributeTypeMap;
    }
}
exports.OptionsOrder = OptionsOrder;
OptionsOrder.discriminator = undefined;
OptionsOrder.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'number',
    },
    {
        name: 'user',
        baseName: 'user',
        type: 'number',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'finishTime',
        baseName: 'finish_time',
        type: 'number',
    },
    {
        name: 'finishAs',
        baseName: 'finish_as',
        type: 'OptionsOrder.FinishAs',
    },
    {
        name: 'status',
        baseName: 'status',
        type: 'OptionsOrder.Status',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'iceberg',
        baseName: 'iceberg',
        type: 'number',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'close',
        baseName: 'close',
        type: 'boolean',
    },
    {
        name: 'isClose',
        baseName: 'is_close',
        type: 'boolean',
    },
    {
        name: 'reduceOnly',
        baseName: 'reduce_only',
        type: 'boolean',
    },
    {
        name: 'isReduceOnly',
        baseName: 'is_reduce_only',
        type: 'boolean',
    },
    {
        name: 'isLiq',
        baseName: 'is_liq',
        type: 'boolean',
    },
    {
        name: 'mmp',
        baseName: 'mmp',
        type: 'boolean',
    },
    {
        name: 'isMmp',
        baseName: 'is_mmp',
        type: 'boolean',
    },
    {
        name: 'tif',
        baseName: 'tif',
        type: 'OptionsOrder.Tif',
    },
    {
        name: 'left',
        baseName: 'left',
        type: 'number',
    },
    {
        name: 'fillPrice',
        baseName: 'fill_price',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'tkfr',
        baseName: 'tkfr',
        type: 'string',
    },
    {
        name: 'mkfr',
        baseName: 'mkfr',
        type: 'string',
    },
    {
        name: 'refu',
        baseName: 'refu',
        type: 'number',
    },
    {
        name: 'refr',
        baseName: 'refr',
        type: 'string',
    },
];
(function (OptionsOrder) {
    let FinishAs;
    (function (FinishAs) {
        FinishAs[FinishAs["Filled"] = 'filled'] = "Filled";
        FinishAs[FinishAs["Cancelled"] = 'cancelled'] = "Cancelled";
        FinishAs[FinishAs["Liquidated"] = 'liquidated'] = "Liquidated";
        FinishAs[FinishAs["Ioc"] = 'ioc'] = "Ioc";
        FinishAs[FinishAs["AutoDeleveraged"] = 'auto_deleveraged'] = "AutoDeleveraged";
        FinishAs[FinishAs["ReduceOnly"] = 'reduce_only'] = "ReduceOnly";
        FinishAs[FinishAs["PositionClosed"] = 'position_closed'] = "PositionClosed";
        FinishAs[FinishAs["ReduceOut"] = 'reduce_out'] = "ReduceOut";
        FinishAs[FinishAs["MmpCancelled"] = 'mmp_cancelled'] = "MmpCancelled";
    })(FinishAs = OptionsOrder.FinishAs || (OptionsOrder.FinishAs = {}));
    let Status;
    (function (Status) {
        Status[Status["Open"] = 'open'] = "Open";
        Status[Status["Finished"] = 'finished'] = "Finished";
    })(Status = OptionsOrder.Status || (OptionsOrder.Status = {}));
    let Tif;
    (function (Tif) {
        Tif[Tif["Gtc"] = 'gtc'] = "Gtc";
        Tif[Tif["Ioc"] = 'ioc'] = "Ioc";
        Tif[Tif["Poc"] = 'poc'] = "Poc";
    })(Tif = OptionsOrder.Tif || (OptionsOrder.Tif = {}));
})(OptionsOrder || (exports.OptionsOrder = OptionsOrder = {}));
