// gate-api/dist/model/optionsPositionClose.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionsPositionClose = void 0;
class OptionsPositionClose {
    static getAttributeTypeMap() {
        return OptionsPositionClose.attributeTypeMap;
    }
}
exports.OptionsPositionClose = OptionsPositionClose;
OptionsPositionClose.discriminator = undefined;
OptionsPositionClose.attributeTypeMap = [
    {
        name: 'time',
        baseName: 'time',
        type: 'number',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'side',
        baseName: 'side',
        type: 'OptionsPositionClose.Side',
    },
    {
        name: 'pnl',
        baseName: 'pnl',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'settleSize',
        baseName: 'settle_size',
        type: 'string',
    },
];
(function (OptionsPositionClose) {
    let Side;
    (function (Side) {
        Side[Side["Long"] = 'long'] = "Long";
        Side[Side["Short"] = 'short'] = "Short";
    })(Side = OptionsPositionClose.Side || (OptionsPositionClose.Side = {}));
})(OptionsPositionClose || (exports.OptionsPositionClose = OptionsPositionClose = {}));
