// gate-api/dist/model/optionsMyTrade.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OptionsMyTrade = void 0;
class OptionsMyTrade {
    static getAttributeTypeMap() {
        return OptionsMyTrade.attributeTypeMap;
    }
}
exports.OptionsMyTrade = OptionsMyTrade;
OptionsMyTrade.discriminator = undefined;
OptionsMyTrade.attributeTypeMap = [
    {
        name: 'id',
        baseName: 'id',
        type: 'number',
    },
    {
        name: 'createTime',
        baseName: 'create_time',
        type: 'number',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'orderId',
        baseName: 'order_id',
        type: 'number',
    },
    {
        name: 'size',
        baseName: 'size',
        type: 'number',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'underlyingPrice',
        baseName: 'underlying_price',
        type: 'string',
    },
    {
        name: 'role',
        baseName: 'role',
        type: 'OptionsMyTrade.Role',
    },
];
(function (OptionsMyTrade) {
    let Role;
    (function (Role) {
        Role[Role["Taker"] = 'taker'] = "Taker";
        Role[Role["Maker"] = 'maker'] = "Maker";
    })(Role = OptionsMyTrade.Role || (OptionsMyTrade.Role = {}));
})(OptionsMyTrade || (exports.OptionsMyTrade = OptionsMyTrade = {}));
