// gate-api/dist/model/positionClose.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PositionClose = void 0;
class PositionClose {
    static getAttributeTypeMap() {
        return PositionClose.attributeTypeMap;
    }
}
exports.PositionClose = PositionClose;
PositionClose.discriminator = undefined;
PositionClose.attributeTypeMap = [
    {
        name: 'time',
        baseName: 'time',
        type: 'number',
    },
    {
        name: 'contract',
        baseName: 'contract',
        type: 'string',
    },
    {
        name: 'side',
        baseName: 'side',
        type: 'PositionClose.Side',
    },
    {
        name: 'pnl',
        baseName: 'pnl',
        type: 'string',
    },
    {
        name: 'pnlPnl',
        baseName: 'pnl_pnl',
        type: 'string',
    },
    {
        name: 'pnlFund',
        baseName: 'pnl_fund',
        type: 'string',
    },
    {
        name: 'pnlFee',
        baseName: 'pnl_fee',
        type: 'string',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
    {
        name: 'maxSize',
        baseName: 'max_size',
        type: 'string',
    },
    {
        name: 'accumSize',
        baseName: 'accum_size',
        type: 'string',
    },
    {
        name: 'firstOpenTime',
        baseName: 'first_open_time',
        type: 'number',
    },
    {
        name: 'longPrice',
        baseName: 'long_price',
        type: 'string',
    },
    {
        name: 'shortPrice',
        baseName: 'short_price',
        type: 'string',
    },
];
(function (PositionClose) {
    let Side;
    (function (Side) {
        Side[Side["Long"] = 'long'] = "Long";
        Side[Side["Short"] = 'short'] = "Short";
    })(Side = PositionClose.Side || (PositionClose.Side = {}));
})(PositionClose || (exports.PositionClose = PositionClose = {}));
