// gate-api/dist/model/spotPriceTrigger.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotPriceTrigger = void 0;
class SpotPriceTrigger {
    static getAttributeTypeMap() {
        return SpotPriceTrigger.attributeTypeMap;
    }
}
exports.SpotPriceTrigger = SpotPriceTrigger;
SpotPriceTrigger.discriminator = undefined;
SpotPriceTrigger.attributeTypeMap = [
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'rule',
        baseName: 'rule',
        type: 'SpotPriceTrigger.Rule',
    },
    {
        name: 'expiration',
        baseName: 'expiration',
        type: 'number',
    },
];
(function (SpotPriceTrigger) {
    let Rule;
    (function (Rule) {
        Rule[Rule["GreaterThanOrEqualTo"] = '>='] = "GreaterThanOrEqualTo";
        Rule[Rule["LessThanOrEqualTo"] = '<='] = "LessThanOrEqualTo";
    })(Rule = SpotPriceTrigger.Rule || (SpotPriceTrigger.Rule = {}));
})(SpotPriceTrigger || (exports.SpotPriceTrigger = SpotPriceTrigger = {}));
