// gate-api/dist/model/spotPricePutOrder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SpotPricePutOrder = void 0;
class SpotPricePutOrder {
    static getAttributeTypeMap() {
        return SpotPricePutOrder.attributeTypeMap;
    }
}
exports.SpotPricePutOrder = SpotPricePutOrder;
SpotPricePutOrder.discriminator = undefined;
SpotPricePutOrder.attributeTypeMap = [
    {
        name: 'type',
        baseName: 'type',
        type: 'SpotPricePutOrder.Type',
    },
    {
        name: 'side',
        baseName: 'side',
        type: 'SpotPricePutOrder.Side',
    },
    {
        name: 'price',
        baseName: 'price',
        type: 'string',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'account',
        baseName: 'account',
        type: 'SpotPricePutOrder.Account',
    },
    {
        name: 'timeInForce',
        baseName: 'time_in_force',
        type: 'SpotPricePutOrder.TimeInForce',
    },
    {
        name: 'autoBorrow',
        baseName: 'auto_borrow',
        type: 'boolean',
    },
    {
        name: 'autoRepay',
        baseName: 'auto_repay',
        type: 'boolean',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
];
(function (SpotPricePutOrder) {
    let Type;
    (function (Type) {
        Type[Type["Limit"] = 'limit'] = "Limit";
        Type[Type["Market"] = 'market'] = "Market";
    })(Type = SpotPricePutOrder.Type || (SpotPricePutOrder.Type = {}));
    let Side;
    (function (Side) {
        Side[Side["Buy"] = 'buy'] = "Buy";
        Side[Side["Sell"] = 'sell'] = "Sell";
    })(Side = SpotPricePutOrder.Side || (SpotPricePutOrder.Side = {}));
    let Account;
    (function (Account) {
        Account[Account["Normal"] = 'normal'] = "Normal";
        Account[Account["Margin"] = 'margin'] = "Margin";
        Account[Account["Unified"] = 'unified'] = "Unified";
    })(Account = SpotPricePutOrder.Account || (SpotPricePutOrder.Account = {}));
    let TimeInForce;
    (function (TimeInForce) {
        TimeInForce[TimeInForce["Gtc"] = 'gtc'] = "Gtc";
        TimeInForce[TimeInForce["Ioc"] = 'ioc'] = "Ioc";
    })(TimeInForce = SpotPricePutOrder.TimeInForce || (SpotPricePutOrder.TimeInForce = {}));
})(SpotPricePutOrder || (exports.SpotPricePutOrder = SpotPricePutOrder = {}));
