// gate-api/dist/model/unifiedLoan.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UnifiedLoan = void 0;
class UnifiedLoan {
    static getAttributeTypeMap() {
        return UnifiedLoan.attributeTypeMap;
    }
}
exports.UnifiedLoan = UnifiedLoan;
UnifiedLoan.discriminator = undefined;
UnifiedLoan.attributeTypeMap = [
    {
        name: 'currency',
        baseName: 'currency',
        type: 'string',
    },
    {
        name: 'type',
        baseName: 'type',
        type: 'UnifiedLoan.Type',
    },
    {
        name: 'amount',
        baseName: 'amount',
        type: 'string',
    },
    {
        name: 'repaidAll',
        baseName: 'repaid_all',
        type: 'boolean',
    },
    {
        name: 'text',
        baseName: 'text',
        type: 'string',
    },
];
(function (UnifiedLoan) {
    let Type;
    (function (Type) {
        Type[Type["Borrow"] = 'borrow'] = "Borrow";
        Type[Type["Repay"] = 'repay'] = "Repay";
    })(Type = UnifiedLoan.Type || (UnifiedLoan.Type = {}));
})(UnifiedLoan || (exports.UnifiedLoan = UnifiedLoan = {}));
